symfony/plugins/orangehrmPimPlugin/Api/EmpEmergencyContactAPI.php

```diff
@@ -0,0 +1,315 @@
+use OrangeHRM\Pim\Api\Model\EmpEmergencyContactModel;
+use OrangeHRM\Pim\Dto\EmpEmergencyContactSearchFilterParams;
+use OrangeHRM\Pim\Service\EmpEmergencyContactService;
+
+class EmpEmergencyContactAPI extends Endpoint implements CrudEndpoint
+{
+    public const PARAMETER_NAME = 'name';
+    public const PARAMETER_RELATIONSHIP = 'relationship';
+    public const PARAMETER_HOME_PHONE = 'homePhone';
+    public const PARAMETER_OFFICE_PHONE = 'officePhone';
+    public const PARAMETER_MOBILE_PHONE = 'mobilePhone';
+
+    public const FILTER_NAME = 'name';
+    public const PARAM_RULE_Max_Length = 100;
+
+    /**
+     * @var EmpEmergencyContactService|null
+     */
+    protected ?EmpEmergencyContactService $empEmergencyContactService = null;
+
+    /**
+     * @return EmpEmergencyContactService
+     */
+    public function getEmpEmergencyContactService(): EmpEmergencyContactService
+    {
+        if (!$this->empEmergencyContactService instanceof EmpEmergencyContactService) {
+            $this->empEmergencyContactService = new EmpEmergencyContactService();
+        }
+        return $this->empEmergencyContactService;
+    }
+
+    /**
+     * @param EmpEmergencyContactService|null $empEmergencyContactService
+     */
+    public function setEmpEmergencyContactService(?EmpEmergencyContactService $empEmergencyContactService): void
+    {
+        $this->empEmergencyContactService = $empEmergencyContactService;
+    }
+
+    /**
+     * @inheritDoc
+     * @throws ServiceException
+     */
+    public function getAll(): EndpointGetAllResult
+    {
+        $emergencyContactSearchParams = new EmpEmergencyContactSearchFilterParams();
+        $empNumber = $this->getRequestParams()->getInt(
+            RequestParams::PARAM_TYPE_ATTRIBUTE,
+            CommonParams::PARAMETER_EMP_NUMBER
+        );
+
+        $this->setSortingAndPaginationParams($emergencyContactSearchParams);
+        $emergencyContactSearchParams->setName(
+            $this->getRequestParams()->getStringOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_NAME
+            )
+        );
+        $emergencyContactSearchParams->setEmpNumber($empNumber);
+        $emergencyContact = $this->getEmpEmergencyContactService()->searchEmployeeEmergencyContacts($emergencyContactSearchParams);
+        return new EndpointGetAllResult(
+            EmpEmergencyContactModel::class, $emergencyContact,
+            new ParameterBag(
+                [
+                    CommonParams::PARAMETER_EMP_NUMBER => $empNumber,
+                    CommonParams::PARAMETER_TOTAL => $this->getEmpEmergencyContactService()->getSearchEmployeeEmergencyContactsCount($emergencyContactSearchParams)
+                ]
+            )
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            $this->getValidationDecorator()->notRequiredParamRule(
```

Comment:
why this rule as not required?

Reply:
yes EMP_number is required, but name filter is not required right ? i did check on postman removing the "?name=first" then i got a error